index/generator/library/library.go

```diff
@@ -74,3 +92,17 @@ func CreateIndexFile(index []schema.Schema, indexFilePath string) error {
 
 	return nil
 }
+
+func validateIndexComponent(indexComponent schema.Schema) error {
```

Comment:
Do we really need to validate the index.json in the index generator itself? I would assume that any index.json it spits out is valid, and we shouldn't need to validate that.

Reply:
IMO, we should at least validate the following things for index.json generation, otherwise the following operations (eg. list stacks, pull down stack) that are using the index.json will be blocked due to missing fields in index.json
- `indexComponent.Name` is not empty so that user can pull down stack resources by choosing the stack name  
- `indexComponent.Links` is not empty so that at least user can pull down the stack devfile (the devfile link is necessary for pulling down the stack devfile)
- `indexComponent.Resources` is not empty because at least the devfile should exist in resources field

Reply:
That's a fair point actually, and it can't hurt to do I suppose.